components/sup/src/manager/service/hook_runner.rs

```diff
@@ -40,7 +57,26 @@ impl<H> HookRunner<H> where H: Hook + Sync
                      pkg,
                      passwd }
     }
+
+    pub fn retriable_future(&self) -> (FutureHandle, impl Future<Item = (), Error = ()>) {
+        let f = future::loop_fn(self.clone(), |hook_runner| {
+            hook_runner.clone()
+                       .into_future()
+                       .map(move |(exit_value, _duration)| {
+                           if <H as Hook>::should_retry(&exit_value) {
```

Comment:
Why not just
```suggestion
                           if H::should_retry(&exit_value) {
```
? It seems to work. Similarly with `file_name`. If you think `H` is unclear, it's fine to be explicit and call it `HOOK`. There's no need for template variables to be limited to single characters.